platform_plugin_turnitin/turnitin.py

```diff
@@ -0,0 +1,427 @@
+            turnitin_submission_id = turnitin_submission.json()["id"]
+            current_user_id = self.get_user_data()["user_id"]
+            current_user = User.objects.get(id=current_user_id)
+            submission = TurnitinSubmission(
+                user=current_user, turnitin_submission_id=turnitin_submission_id
+            )
+            submission.save()
+            myfile = data.params["myfile"].file
+            # turnitin_submission_id='0a966646-83f9-4ce6-aa47-71e07baf4e30'
+            response = put_upload_submission_file_content(
+                turnitin_submission_id, myfile
+            )
+            return Response(
+                json.dumps(response.json()),
+                content_type="application/json",
+                charset="UTF-8",
+            )
+        return Response(
+            json.dumps(turnitin_submission.json()),
+            content_type="application/json",
+            charset="UTF-8",
+        )
+
+    @XBlock.json_handler
+    def get_submission_status(self, data, suffix=""):
+        """
+        Retrieves the status of the latest Turnitin submission for the user.
+
+        Args:
+            data (dict): Input data for the request.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            dict: Information related to the user's latest Turnitin submission.
+        """
+        current_user_id = self.get_user_data()["user_id"]
+        current_user = User.objects.get(id=current_user_id)
+        try:
+            last_submission = TurnitinSubmission.objects.filter(
+                user=current_user
+            ).latest("created_at")
+        except TurnitinSubmission.DoesNotExist:
+            return {"success": False}
+        # last_submission = 'de6784c5-471f-4220-aff1-16b6b44dffcf'
+        response = get_submission_info(last_submission.turnitin_submission_id)
+        return response.json()
+
+    @XBlock.json_handler
+    def generate_similarity_report(self, data, suffix=""):
+        """
+        Initiates the generation of a similarity report for the user's latest Turnitin submission.
+
+        Args:
+            data (dict): Input data for the request.
+            suffix (str, optional): Additional suffix for the request. Defaults to ''.
+
+        Returns:
+            dict: The status of the similarity report generation process.
+        """
+
+        payload = {
+            "indexing_settings": {"add_to_index": True},
+            "generation_settings": {
+                "search_repositories": [
+                    "INTERNET",
+                    "SUBMITTED_WORK",
+                    "PUBLICATION",
+                    "CROSSREF",
+                    "CROSSREF_POSTED_CONTENT",
+                ],
+                "submission_auto_excludes": [
+                    "b84b77d1-da0f-4f45-b002-8aec4f4796d6",
+                    "b86de142-bc44-4f95-8467-84af12b89217",
```

Comment:
what's this for?

Reply:
@mariajgrimaldi The payloads contain a series of optional parameters; that's a standard request for the purposes of this test.

Reply:
I meant what are those random ids from `submission_auto_excludes`? Do they mean something? Where did we get them?